File: TestDome/FileOwners.py
class FileOwners:

    @staticmethod
    def group_by_owners(files):
        r = {}
        for file_name in files:
            r.setdefault(files[file_name], [])
            r[files[file_name]].append(file_name)
        return r

files = {
    'Input.txt': 'Randy',
    'Code.py': 'Stan',
    'Output.txt': 'Randy'
}
print(FileOwners.group_by_owners(files))

File: TestDome/FileOwners.php
<?php
class FileOwners
{
    public static function groupByOwners($files)
    {
    	$groupedByOwner = array();
        foreach ($files as $file => $owner) {
        	if (!array_key_exists($owner, $groupedByOwner)) {
        		$groupedByOwner[$owner] = array();
        	}

        	$groupedByOwner[$owner][] = $file;
        }

        return $groupedByOwner;
    }
}

$files = array
(
    "Input.txt" => "Randy",
    "Code.py" => "Stan",
    "Output.txt" => "Randy"
);
var_dump(FileOwners::groupByOwners($files));